EIPS/pay_req_url_fmt.md

```diff
@@ -0,0 +1,65 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: URL Format for Transaction Requests
+    Author: Daniel A. Nagy <daniel@ethereum.org>
+    Type: Standard Track
+    Category: ERC
+    Status: Draft
+    Replaces: 67
+    Created: 2017-08-01
+    Requires: 20
+
+## Simple Summary
+A standard way of representing various transactions, especially payment requests in Ethers and ERC #20 tokens as URLs.
+
+## Abstract
+URLs embedded in QR-codes, hyperlinks in web-pages, emails or chat messages provide for robust cross-application signaling between very loosely coupled applications. A standardized URL format for payment requests allows for instant invocation of the user's preferred wallet application (even if it is a webapp or a swarm đapp), with the correct parameterization of the payment transaction only to be confirmed by the (authenticated) user.
+
+## Motivation
+The convenience of representing payment requests by standard URLs has been a major factor in the wide adoption of Bitcoin. Bringing a similarly convenient mechanism to Ethereum would speed up its acceptance as a payment platform among end-users. In particular, URLs embedded in broadcast Intents are the preferred way of launching applications on the Android operating system and work across practically all applications. Desktop web browsers have a standardized way of defining protocol handlers for URLs with specific protocol specifications. Other desktop applications typically launch the web browser upon encountering a URL. Thus, payment request URLs could be delivered through a very broad, ever growing selection of channels.
+
+This specification supersedes ERC #67, which is a URL format for representing arbitrary transactions in a low-level fashion. This ERC focuses specifically on the important special case of payment requests, while allowing for other, ABI-specified transactions.
+
+## Specification
+
+### Syntax
+Payment request URLs contain "ethereum" in their schema (protocol) part and are constructed as follows:
+
+    request                 = "ethereum" ":" target_address [ "/" function_name ] [ "?" parameters ]
+    target_address          = ethereum_address
+    function_name           = STRING
+    ethereum_address        = ( "0x" 40*40HEXDIG ) / ENS_NAME
+    parameters              = parameter *( "&" parameter )
+    parameter               = key "=" value
+    key                     = "value" / "gas" / TYPE
```

Comment:
I would add another important optional parameter: 

```
"value" / "gas" / "callback" / TYPE
```

Callback calls the name javascript function when the transaction is approved or rejected by the client, with the parameters (error, result). In the result will be the transaction ID, so that the app can handle it however it wants.

Reply:
I agree with adding `gas`, but would appreciate if you could share a compelling use case for `callback` and an example.

Reply:
Callback is used on almost all apps. It's how the app can react to the user knowing that it has paid. When an app asks the user for a transaction, without a callback, not only it has to wait until the transaction is confirmed, it has no way to link any particular user to a particular transaction. This will lead to apps that do not respond for up to a minute or more (depending on the gas price paid) and has a much more complex interface, leading to a bad user experience.

With a callback, as soon as the user types their password, sends the transaction and goes back to the app, the app can immediately respond, by showing them a screen saying "your transaction has been made, just waiting for confirmations", and saving the data locally if needed.

Callbacks are used in about any app that uses an async method and are super important.

Reply:
@alexvandesande Just to make sure that I understand you correctly, let me make up the example for you. Use case: buying air tickets from a mobile dapp. The payment request would look something like this:
```ethereum:payments.airtickets.eth?value=10.31e18&callback=bzz%3A%2F%2Fairtickets.eth%2Forder%2FG34DF5```
Upon clicking on this link, the user's preferred payment app would come up and upon composing and submitting the transaction, it would redirect the user to the corresponding order page, which might be caught as an intent by the airtickets mobile app or, in its absence, redirect to the corresponding swarm site.

Now, in order for this to be usable, it should probably have some wildcards for inserting the transaction id or even some indication that the payment request was declined. That opens a whole new can of worms. I would suggest postponing this to some future version, since apparently BIP 20 does not have such a capability and yet the Bitcoin ecosystem makes do without it.

Reply:
You're correct. I was thinking about Mist/Metamask specific callbacks, in which the client has a direct access to calling javascript functions inside the running app, but this would require more complexity.

I'm ok with moving the callback in a future EIP